certification/code/certificates.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+'''
+Generate a certificate from a template.
+
+* Requires the python package 'cairosvg' to be installed.
+  Please visit http://cairosvg.org/ for install instructions.
+* Some systems may also need to have 'cairo' installed.
+  Please visit http://cairographics.org/download/ for the same.
+'''
+
+import sys
+import os
+import re
+import tempfile
+import cairosvg
+from django.conf import settings
+
+#---------------------------------------------------------------
+# Main Functions
+
+def generate(args):
+    '''Process a single entry and returns their certificate'''
+    root_dir = os.path.dirname(__file__)
+    template_path = construct_template_path(root_dir, args.badge_type)
+    return create_certificate(template_path, args.params, args.cert_id)
+
+
+def create_certificate(template_path, params, cert_id):
+    '''Creates and returns a single certificate.'''
+
+    with open(template_path, 'r') as reader:
+        template = reader.read()
+    check_template(template, params)
+
+    for key, value in params.items():
+        pattern = '{{' + key + '}}'
+        template = template.replace(pattern, value)
+
+    tmp = tempfile.NamedTemporaryFile(suffix='.svg', delete=False)
+    # tmp = open(str(params.id) + '.svg', 'wb')
+    tmp.write(bytes(template, 'utf-8'))
+    filename = os.path.join(settings.CERTIFICATES_DIR, str(cert_id) + '.pdf')
+    cairosvg.svg2pdf(url=tmp.name, dpi=90, write_to=filename)
+    return
```

Comment:
Yes, `svg2pdf` returns `None` if the `write_to` argument is supplied. And this is what the docs say:
>If the `write_to` argument is provided (filename or file-like object), the output is written there. Otherwise, the function returns a byte string.